lib/App/Yath/Command/runner.pm

```diff
@@ -56,7 +57,9 @@ sub generate_run_sub {
     my $runner_pid = $$;
     my $jump = setjump "Test-Runner" => sub {
         local $.;
-        local %SIG = %SIG;
+        my @SIGNAMES = grep { $_ ne 'ZERO' } split /\s+/, $Config{sig_name};
+        local @SIG{ @SIGNAMES } = @SIG{ @SIGNAMES };
```

Comment:
will local `$SIG{FOO} = $SIG{FOO}` will cause the 'FOO' key to exist where it previously did not, default value of undef. Does the value 'undef' have special meaning in %SIG? I am worried the 'undef' might override a default-to-ignore or similar.

I am wondering if we should skip using `local` and instead just copy %SIG, then copy it back and delete any new keys that were added since the copy was made.

Reply:
to be honest in the context of yath, we should probably consider to simply remove this line... 
why do we want to localize the signal here?

by default all signals exist and have some magic attached to them
are you worried about using a signal KEY (like FOO) not already defined?

What you are suggesting could work, but same as shown here you would have to make sure to define them in the correct order... 

The problem is for duplicate signals the first defined give its name to the signal.. 
So when setting/restoring them you need to be careful


Note: we are working on Perl patch to try to solve the issue at the root

Reply:
Note: I think the integration test is good to keep as it shows the regression / different behavior introduced and we want to avoid it if possible...

Reply:
This is because the runner itself sets up signal handlers, and we want to be SURE they are cleared before any tests are run in the processes forked from the runner, which is how tests are run by default.

Reply:
ok, I still think that in this case the current patch is the best solution.
We are working for a Perl patch, but this will only make it to 5.32
https://github.com/Perl/perl5/pull/17509

Reply:
> Does the value 'undef' have special meaning in %SIG? I am worried the 'undef' might override a default-to-ignore or similar.

Undef has a distinct side effect: it disables the C-level signal handler. That's exactly why `local %SIG` doesn't dwim, on restoration the correct perl-level handler is restored but the c-level handler isn't (as restoration doesn't trigger set magic)